src/WebJobs.Script/OutOfProc/Http/DefaultHttpWorkerService.cs

```diff
@@ -0,0 +1,205 @@
+        {
+            _logger.LogDebug("Will invoke simple httpTrigger function:{functionName} invocationId:{invocationId}", scriptInvocationContext.FunctionMetadata.Name, scriptInvocationContext.ExecutionContext.InvocationId);
+
+            ScriptInvocationResult scriptInvocationResult = new ScriptInvocationResult()
+            {
+                Outputs = new Dictionary<string, object>(),
+                Return = new object()
+            };
+            HttpRequestMessage httpRequestMessage = null;
+            var input = scriptInvocationContext.Inputs.FirstOrDefault();
+            HttpRequest httpRequest = (HttpRequest)input.val;
+            if (httpRequest == null)
+            {
+                throw new ArgumentNullException(nameof(httpRequest));
+            }
+            try
+            {
+                // Build HttpRequestMessage from HttpTrigger binding
+                HttpRequestMessageFeature httpRequestMessageFeature = new HttpRequestMessageFeature(httpRequest.HttpContext);
+                httpRequestMessage = httpRequestMessageFeature.HttpRequestMessage;
+
+                AddRequestHeadersAndSetRequestUri(httpRequestMessage, scriptInvocationContext.FunctionMetadata.Name, scriptInvocationContext.ExecutionContext.InvocationId.ToString());
+
+                // Populate query params from httpTrigger
+                string httpInvokerUri = QueryHelpers.AddQueryString(httpRequestMessage.RequestUri.ToString(), HttpRequestMessageConverters.ConvertQueryCollectionToDictionary(httpRequest.Query));
+                httpRequestMessage.RequestUri = new Uri(httpInvokerUri);
+
+                _logger.LogDebug("Sending http request message for simple httpTrigger function:{functionName} invocationId:{invocationId}", scriptInvocationContext.FunctionMetadata.Name, scriptInvocationContext.ExecutionContext.InvocationId);
+
+                HttpResponseMessage invocationResponse = await _httpClient.SendAsync(httpRequestMessage);
+
+                _logger.LogDebug("Received http response for simple httpTrigger function:{functionName} invocationId:{invocationId}", scriptInvocationContext.FunctionMetadata.Name, scriptInvocationContext.ExecutionContext.InvocationId);
+
+                BindingMetadata httpOutputBinding = scriptInvocationContext.FunctionMetadata.OutputBindings.FirstOrDefault();
+                if (httpOutputBinding != null)
+                {
+                    // handle http output binding
+                    scriptInvocationResult.Outputs.Add(httpOutputBinding.Name, invocationResponse);
+                    // handle $return
+                    scriptInvocationResult.Return = invocationResponse;
+                }
+                scriptInvocationContext.ResultSource.SetResult(scriptInvocationResult);
+            }
+            catch (Exception responseEx)
+            {
+                scriptInvocationContext.ResultSource.TrySetException(responseEx);
```

Comment:
this will get logged properly upstream?

Reply:
correct. This will show up as FunctionInvocationException